tests/unit/embeddings/test_mistralai_embedder.py

```diff
@@ -0,0 +1,40 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+from neo4j_graphrag.embeddings.mistral import MistralAIEmbeddings
+
+
+@patch("neo4j_graphrag.embeddings.mistral.Mistral", None)
+def test_mistralai_embedder_missing_dependency() -> None:
+    with pytest.raises(ImportError):
+        MistralAIEmbeddings()
+
+
+@patch("neo4j_graphrag.embeddings.mistral.Mistral")
+def test_mistralai_embedder_happy_path(mock_mistralai: Mock) -> None:
+    mock_mistral_instance = mock_mistralai.return_value
```

Comment:
Maybe a personal preference, commenting just in case (do not change anything if you prefer the way it is now): we can chain this in one single line like this (well, example with OpenAI):
```
mock_openai.return_value.chat.completions.create.return_value = MagicMock(
        choices=[MagicMock(message=MagicMock(content="openai chat response"))],
    )
```
(I haven't tested with async, that will be a nice test)

Reply:
Hmm I think I personally prefer it to not be a one-liner if it's quite heavily nested. I'll keep it this way if alright with you